packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -229,7 +229,39 @@ async function generateContentLengthHeader(config: AxiosRequestConfig) {
 	}
 }
 
-async function parseRequestObject(requestObject: IDataObject) {
+type RequestObject = Partial<{
```

Comment:
A lot of the nodes currently use `OptionsWithUri` for some of these fields. Maybe we can use `IHttpRequestOptions` interface from the `workflow` package, and also update at least one node to use that interface instead of `OptionsWithUri`.
Then this type can extend the base one with all the additional fields, that we don't want to let nodes pass in.

Reply:
Yep I saw that `IHttpRequestOptions` type, but there are some fields missing/different than what's being used in the code. Should I update `IHttpRequestOptions` with the missing fields and just use that one?


Side by side:
```
type RequestObject = Partial<{
	baseURL: string;
	uri: string;
	url: string;
	method: Method;
	qs: Record<string, unknown>;
	qsStringifyOptions: { arrayFormat: 'repeat' | 'brackets' };
	useQuerystring: boolean;
	headers: Record<string, string>;
	auth: Partial<{
		sendImmediately: boolean;
		bearer: string;
		user: string;
		username: string;
		password: string;
		pass: string;
	}>;
	body: unknown;
	formData: FormData;
	form: FormData;
	json: boolean;
	useStream: boolean;
	encoding: string;
	followRedirect: boolean;
	followAllRedirects: boolean;
	timeout: number;
	rejectUnauthorized: boolean;
	proxy: string | AxiosProxyConfig;
	simple: boolean;
	resolveWithFullResponse: boolean;
}>;
```

```
export interface IHttpRequestOptions {
	url: string;
	baseURL?: string;
	headers?: IDataObject;
	method?: IHttpRequestMethods;
	body?: FormData | GenericValue | GenericValue[] | Buffer | URLSearchParams;
	qs?: IDataObject;
	arrayFormat?: 'indices' | 'brackets' | 'repeat' | 'comma';
	auth?: {
		username: string;
		password: string;
		sendImmediately?: boolean;
	};
	disableFollowRedirect?: boolean;
	encoding?: 'arraybuffer' | 'blob' | 'document' | 'json' | 'text' | 'stream';
	skipSslCertificateValidation?: boolean;
	returnFullResponse?: boolean;
	ignoreHttpStatusErrors?: boolean;
	proxy?: {
		host: string;
		port: number;
		auth?: {
			username: string;
			password: string;
		};
		protocol?: string;
	};
	timeout?: number;
	json?: boolean;
}
```

Reply:
TBH, I'm not sure about what exactly needs to happen here. I'd just like to see a unified type for request options object, and preferably in the `workflow` package, so that we can make all request configs consistently typed in the nodes code.
If there should be some one interface or two, is up to you.

Reply:
Yep I'm also not sure yet, will look into it tomorrow